### DISCUSSION TITLE
How to properly configure `no-undef-components`?

### DISCUSSION THREAD

**Discussion by pixunil** [nuxt/eslint-config#521] [Q&A]
Hi!

the rule [`no-undef-components`](https://eslint.vuejs.org/rules/no-undef-components.html) is really helpful in spotting some mistakes (e.g. typos, eagerly renaming stuff), but as its description states, it can’t work with globally defined components out of the box; and so auto-imports are also hidden.

In Nuxt, there is `.nuxt/components.d.ts` which lists all components, and defines a constant `componentNames`, which I thought could be used as allow list. However, I could not import such a constant or find an appropriate function in Nuxt Kit to do something similar.

Is there a possibility to configure `no-undef-components` such that it works with auto-imported components too?

**Comment by elliottregan** [comment]
I'm struggling with this, too. I'm currently manually adding the list of components I use, but that doesn't seem right to me.

**Reply by elliottregan** [reply]
This might be a better question for the Nuxt community, though.
